In [1]:
import pandas as pd

d = pd.read_csv("D:\data_science\pandas\stocks.csv")

print(type(d))
print(d.shape)
print()
print(d.head())    # gives the top 5 rows of data
print()
print(d.tail())    # gives the bottom 5 rows of the data
print()
print(d.head(11))  #  --> d.head(no. of rows)
print()
print(d.tail(9))   #  --> d.tail(no. of rows)

<class 'pandas.core.frame.DataFrame'>
(3313, 7)

         Date       Open       High        Low      Close  Adj Close    Volume
0  2004-08-19  49.676899  51.693783  47.669952  49.845802  49.845802  44994500
1  2004-08-20  50.178635  54.187561  49.925285  53.805050  53.805050  23005800
2  2004-08-23  55.017166  56.373344  54.172661  54.346527  54.346527  18393200
3  2004-08-24  55.260582  55.439419  51.450363  52.096165  52.096165  15361800
4  2004-08-25  52.140873  53.651051  51.604362  52.657513  52.657513   9257400

            Date        Open        High         Low       Close   Adj Close  \
3308  2017-10-09  980.000000  985.424988  976.109985  977.000000  977.000000   
3309  2017-10-10  980.000000  981.570007  966.080017  972.599976  972.599976   
3310  2017-10-11  973.719971  990.710022  972.250000  989.250000  989.250000   
3311  2017-10-12  987.450012  994.119995  985.000000  987.830017  987.830017   
3312  2017-10-13  992.000000  997.210022  989.000000  989.679993  989.679993

In [4]:
import pandas as pd

d = pd.read_csv("D:\data_science\pandas\stocks.csv")

print(d.columns)   # gives list of column names
print()
print(d)           # as it is large dataset, we cannot add index no./index lables to the dataset. so, python adds index no. for each rows

Index(['Date', 'Open', 'High', 'Low', 'Close', 'Adj Close', 'Volume'], dtype='object')

            Date        Open        High         Low       Close   Adj Close  \
0     2004-08-19   49.676899   51.693783   47.669952   49.845802   49.845802   
1     2004-08-20   50.178635   54.187561   49.925285   53.805050   53.805050   
2     2004-08-23   55.017166   56.373344   54.172661   54.346527   54.346527   
3     2004-08-24   55.260582   55.439419   51.450363   52.096165   52.096165   
4     2004-08-25   52.140873   53.651051   51.604362   52.657513   52.657513   
...          ...         ...         ...         ...         ...         ...   
3308  2017-10-09  980.000000  985.424988  976.109985  977.000000  977.000000   
3309  2017-10-10  980.000000  981.570007  966.080017  972.599976  972.599976   
3310  2017-10-11  973.719971  990.710022  972.250000  989.250000  989.250000   
3311  2017-10-12  987.450012  994.119995  985.000000  987.830017  987.830017   
3312  2017-10-13  992.000000  99

In [5]:
import pandas as pd

d = pd.read_csv("D:\data_science\pandas\stocks.csv")

print(d.isnull())              # applies isnull() method to every value in DataFrame and gives bool
print(d.isnull().sum())        # sums no. of NaN in each column
print(d.isnull().sum().sum())  # sums total no. of NaN in DataFrame [or] sums above 1 column values and give result
print()
print(d.isnull().any())        # any() --> tells if any value in columns are missing(NaN) and gives Bool [But use sum(), as it 
                               # tells how many are missing also ]

       Date   Open   High    Low  Close  Adj Close  Volume
0     False  False  False  False  False      False   False
1     False  False  False  False  False      False   False
2     False  False  False  False  False      False   False
3     False  False  False  False  False      False   False
4     False  False  False  False  False      False   False
...     ...    ...    ...    ...    ...        ...     ...
3308  False  False  False  False  False      False   False
3309  False  False  False  False  False      False   False
3310  False  False  False  False  False      False   False
3311  False  False  False  False  False      False   False
3312  False  False  False  False  False      False   False

[3313 rows x 7 columns]
Date         0
Open         0
High         0
Low          0
Close        0
Adj Close    0
Volume       0
dtype: int64
0

Date         False
Open         False
High         False
Low          False
Close        False
Adj Close    False
Volume       False
dtype: bool


In [6]:
import pandas as pd

d = pd.read_csv("D:\data_science\pandas\stocks.csv")

print(d.describe())    # describe() --> gives statistical information of each columns
print()
print(d["Low"].describe())    # can do for selected column
print()
print(d[["High","Open"]].describe())   # also more than 1 column by making argument into 1 list [as argument should be 1 for d[]]

              Open         High          Low        Close    Adj Close  \
count  3313.000000  3313.000000  3313.000000  3313.000000  3313.000000   
mean    380.186092   383.493740   376.519309   380.072458   380.072458   
std     223.818650   224.974534   222.473232   223.853780   223.853780   
min      49.274517    50.541279    47.669952    49.681866    49.681866   
25%     226.556473   228.394516   224.003082   226.407440   226.407440   
50%     293.312286   295.433502   289.929291   293.029114   293.029114   
75%     536.650024   540.000000   532.409973   536.690002   536.690002   
max     992.000000   997.210022   989.000000   989.679993   989.679993   

             Volume  
count  3.313000e+03  
mean   8.038476e+06  
std    8.399521e+06  
min    7.900000e+03  
25%    2.584900e+06  
50%    5.281300e+06  
75%    1.065370e+07  
max    8.276810e+07  

count    3313.000000
mean      376.519309
std       222.473232
min        47.669952
25%       224.003082
50%       289.929291
75%     

**In above Output, there are 25% values which are below 226 no. and 50% values which are below 293 and 75% values which are below 536. So, and when we see min and max, and 50% values are below 300, we can say there are some outliers present in the data. By visualising through graph, we can identify those outliers.**

In [7]:
import pandas as pd

d = pd.read_csv("D:\data_science\pandas\stocks.csv")

print(d.max())
print()
print(d["Close"].min())    # can do methods for each column [or] a list of columns

Date         2017-10-13
Open              992.0
High         997.210022
Low               989.0
Close        989.679993
Adj Close    989.679993
Volume         82768100
dtype: object

49.681866


Multi- Collinearity: relation between columns. Range --> (-1,1). if the value is towards 0 to 1, then they are positive correlated (proportional). If the value is 0, then they are not impacting eachother (unrelated). If the value is towards 0 to -1, then they are negative correlated (inverse proportional).

Now,
Independent variable --> varaibles/columns which are used to find the dependent variables/column through multi collinearity [column][row]
Dependent variable --> variable/column which is need to be predicted through Independent Variable/columns. [row][column]

IMPORTANT POINTS:
1) If the correlation between 2 Independent features/variables is high (negative/positive) it will introduce multi-collinearity in the data. In this case, we should remove one of them.
2) If the correlation between a Independent feature/variable and a Dependent feature/variable is high then we consider that this Independent feature is very important to the model.
3) If a Independent featureis having 0 correlation with the dependent feature then it is considered it is not at all important in the study or for the model and can be dropped.

In [5]:
import pandas as pd

d = pd.read_csv("D:\data_science\pandas\stocks.csv")

# gives the correlation between the each rows and each columns for only which has no strings.
print(d[["Open","High","Low","Close","Adj Close","Volume"]].corr())    

               Open      High       Low     Close  Adj Close    Volume
Open       1.000000  0.999904  0.999845  0.999745   0.999745 -0.564258
High       0.999904  1.000000  0.999834  0.999868   0.999868 -0.562749
Low        0.999845  0.999834  1.000000  0.999899   0.999899 -0.567007
Close      0.999745  0.999868  0.999899  1.000000   1.000000 -0.564967
Adj Close  0.999745  0.999868  0.999899  1.000000   1.000000 -0.564967
Volume    -0.564258 -0.562749 -0.567007 -0.564967  -0.564967  1.000000
